packages/core/src/middleware/defaults/bypassPaymasterMiddleware.ts

```diff
@@ -0,0 +1,29 @@
+import type { UserOperationOverrides, UserOperationStruct } from "../../types";
+import type { ClientMiddlewareFn } from "../types";
+
+export const bypassPaymasterMiddleware: ClientMiddlewareFn = async (
```

Comment:
why not just use the `noopMiddleware`? you're already checking if `paymaster` === `0x` before this ever gets called so this is not doing anything extra

Reply:
> specifically, the paymaster gas limits that were supplied from the gas estimator.

if the override paymaster field is `0x` then the gas estimator shouldn't be returning any paymaster fields. The middleware should result in no paymaster code running anywhere and no paymaster fields being set earlier in the pipeline. Noop should work here based on that

Reply:
yes it could, now that gas Estimator response includes paymasterVerificationGas, and this should be omitted for bypassing paymaster.

Thus, bypassmiddleware is not the same as noopmiddleware.

If paymaster is configured, gas Estimator response WILL include the paymasterVerificationGas field.

Reply:
ditto: my question to @dphilipson 

"okay, @dphilipson just to confirm, if paymaster address is not set on the user op struct sent for estimate user operation gas RPC, the bundler will never send the paymaster gas limits? Not just rundler, but all other bundlers. This is not clear on the 4337 standards."